admin/class-boldgrid-backup-admin-environment.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * File: class-boldgrid-backup-admin-environment.php
+ *
+ * @link https://www.boldgrid.com
+ * @since SINCEVERSION
+ *
+ * @package    Boldgrid_Backup
+ * @subpackage Boldgrid_Backup/admin
+ * @copyright  BoldGrid
+ * @version    $Id$
+ * @author     BoldGrid <support@boldgrid.com>
+ */
+
+/**
+ * Class: Boldgrid_Backup_Admin_Environment
+ *
+ * The purpose of this class is to help us get more information about our environment. It should also
+ * help answer questions, such as, has our environment changed?
+ *
+ * @since SINCEVERSION
+ */
+class Boldgrid_Backup_Admin_Environment {
+	/**
+	 * The option to which we'll save environment data to.
+	 *
+	 * For example, to know if something has changed in the environment, we'll need to know both the
+	 * old values and the new values. The old values will be stored in this option.
+	 *
+	 * @since SINCEVERSION
+	 * @access private
+	 * @var array
+	 */
+	private $option_name = 'boldgrid_backup_environment';
+
+	/**
+	 * Get a saved environment value.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @see self::option_name
+	 *
+	 * @param  string $key     The name of the value to get.
+	 * @param  mixed  $default The default value to return.
+	 * @return mixed
+	 */
+	private function get_saved_value( $key, $default = false ) {
+		$saved_values = $this->get_saved_values();
+
+		return isset( $saved_values[ $key ] ) ? $saved_values[ $key ] : $default;
+	}
+
+	/**
+	 * Get all of our saved values.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @see self::option_name
+	 *
+	 * @return array
+	 */
+	private function get_saved_values() {
+		return get_option( $this->option_name, array() );
+	}
+
+	/**
+	 * Save a specific environment value.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @see self::option_name
+	 *
+	 * @param string $key   The environment key, like "hostname".
+	 * @param mixed  $value The value, such as "domain.com".
+	 */
+	private function save_value( $key, $value ) {
+		$saved_values = $this->get_saved_values();
+
+		$saved_values[ $key ] = $value;
+
+		$this->save_values( $saved_values );
+	}
+
+	/**
+	 * Save all of our environment values.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @see self::option_name
+	 *
+	 * @param array $values All of our environment values.
+	 */
+	private function save_values( $values ) {
```

Comment:
If this function is private, should you add the @access private to the docblock? Not sure based on the documentation if it should or not. I was always under the impression that private methods should have this, as well as private properties.
(https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#1-functions-class-methods)

Reply:
fixed.